kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala

```diff
@@ -58,6 +58,10 @@ private[v1] class BatchesResource extends ApiRequestContext with Logging {
   private lazy val internalConnectTimeout =
     fe.getConf.get(BATCH_INTERNAL_REST_CLIENT_CONNECT_TIMEOUT).toInt
 
+  private def batchV2Enabled(reqConf: Map[String, String]): Boolean = {
+    reqConf.getOrElse(BATCH_API_VERSION.key, fe.getConf.get(BATCH_API_VERSION)) == "2"
```

Comment:
is it a server side behavior?

Reply:
Yes, it's allowed to explicitly enable batch v2 at the app level on the client side.

Reply:
so, in server side, the batch v2 is always supported?

Reply:
I suppose `kyuubi.batch.submitter.enabled` will be removed in the future, and always be enabled when the REST frontend is enabled.

Reply:
what would happen if client enable batch v2 but it is not supported in server side?

Reply:
When the server sets `kyuubi.batch.submitter.enabled=false`, and the client creates a batch with `kyuubi.batch.api.version=2`, the server would return an error.

Reply:
I think this config item is confused. If I am a customer, I may think that the api version here is rest api version /api/v1 or /api/v2.

Reply:
yea, it's a valid concern, I planned to use `/api/v2` at the beginning, but during the implementation, I found there is nothing change in the user API, so I choose to introduce a parameter instead of bumping the API version.

Any suggestion for the name here? `kyuubi.batch.implementation=1,2`?

Reply:
Prefer: kyuubi.batch.impl.version=1,2

Reply:
sgtm, will change soon.